libkatana_python_native/src/PropertyGraph.cpp

```diff
@@ -1030,11 +1035,11 @@ DefPropertyGraph(py::module& m) {
 
   cls.def(
       "write",
-      [](PropertyGraph& self, const std::string* path,
+      [](PropertyGraph& self, const katana::URI& path,
          const std::string& provenance, TxnContext* txn_ctx) {
         TxnContextArgumentHandler txn_handler(txn_ctx);
-        if (path) {
-          return self.Write(*path, provenance, txn_handler.get());
+        if (!path.empty()) {
+          return self.Write(path, provenance, txn_handler.get());
```

Comment:
The idea of "empty" objects doesn't really make any sense in python. So we'll take either a real value or `None`. Pybind11 is nice enough to convert `None` to `null` if you take a pointer instead of a reference, so that's the way to go.

Reply:
The way I think about empty object is that they are "null" in some sense. That means that any value that allows this is implicitly `std::optional<T>` even when written `T`. In some cases it's better to think of the "empty" value as `0`, but this has the problem that `0` is in fact a valid value, and often the empty value is not actually valid.

Reply:
Hmm.  This used to be a pointer to an RDG location and the check was whether the pointer was null.

I changed it to a reference to a URI.  I figure if the URI is empty, then we commit rather than write to a new location.  I'm not sure what your suggestion is.  Only pass a pointer to a URI to maintain the pointer check?  `empty()` is defined on URIs and makes sense to me as something to check to see if we are specifying a new location for the RDG.

Reply:
In C++ this is fine. It doesn't make sense in Python because everything is always a "pointer" (actually a reference, but that term is overloaded in C++). And this is Python interfacing code, so we need to handle things the Pythonic way.